public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -91,19 +91,21 @@ export function areQueriesTransformableToSimpleCondition(
   if (dataQueries.length !== 1) {
     return false;
   }
+  const reducerRemovedOk =
+    'instant' in dataQueries[0].model && dataQueries[0].model.instant && expressionQueries.length === 1;
```

Comment:
Is it sufficient to just check for the expressions length here? Should we not assert that the expression is a threshold expression?

Reply:
yes, it's sufficient because in this [line](https://github.com/grafana/grafana/pull/94246/files/b63dbf7a4fd8a27ef961559ff7ab383ef7cbd38f#diff-6c9d7793ae5d088b056f743df107749d54dd7768b4744d97d1131c2000d650cdR125), we add this expression to ` Boolean(thresholdOk)` => this means that we have only one expression, query is instant, and the only expression is a reducer:

`return (Boolean(reduceOk) || Boolean(reducerRemovedOk)) && Boolean(thresholdOk);`